### PR TITLE
fix reshape issues

### PR DESCRIPTION
**Pull request by kozakromch**
https://github.com/stack-of-tasks/pinocchio/issues/1481.

Thank you for providing the source code. I hope this code will work correctly

### CODE REVIEW

**Comment by hrp2-14** [comment]
Hi ! This project doesn't usually accept pull requests on master. If this wasn't intentionnal, you
can change the base branch of this pull request to devel (No need to close it for that). Best, a bot.

**Review by jcarpent — APPROVED** [review]
Thanks a lot for this PR @kozakromch. This is very helpful.

File: doc/d-practical-exercises/src/robot_hand.py [hunk 1]
```diff
@@ -44,19 +44,18 @@ def collision(self, c2, data=None, oMj1=None, oMj2=None):
...
-        a1 = M1.act(np.matrix([0, 0, -l1 / 2]).T)
-        b1 = M2.act(np.matrix([0, 0, -l2 / 2]).T)
-        a2 = M1.act(np.matrix([0, 0, +l1 / 2]).T)
-        b2 = M2.act(np.matrix([0, 0, +l2 / 2]).T)
-
-        ab = pinv(np.hstack([a1 - a2, b2 - b1])) * (b2 - a2)
+        a1 = M1.act(np.array([0, 0, -l1 / 2]).T)
+        b1 = M2.act(np.array([0, 0, -l2 / 2]).T)
+        a2 = M1.act(np.array([0, 0, +l1 / 2]).T)
+        b2 = M2.act(np.array([0, 0, +l2 / 2]).T)
```
Comment (jmirabel) [inline doc/d-practical-exercises/src/robot_hand.py:56]: I think the `.T` doesn't do anything.
```suggestion
        a1 = M1.act(np.array([0, 0, -l1 / 2]))
        b1 = M2.act(np.array([0, 0, -l2 / 2]))
        a2 = M1.act(np.array([0, 0, +l1 / 2]))
        b2 = M2.act(np.array([0, 0, +l2 / 2]))
```

File: doc/d-practical-exercises/src/robot_hand.py [hunk 2]
```diff
@@ -44,19 +44,18 @@ def collision(self, c2, data=None, oMj1=None, oMj2=None):
...
-        a2 = M1.act(np.matrix([0, 0, +l1 / 2]).T)
-        b2 = M2.act(np.matrix([0, 0, +l2 / 2]).T)
-
-        ab = pinv(np.hstack([a1 - a2, b2 - b1])) * (b2 - a2)
+        a1 = M1.act(np.array([0, 0, -l1 / 2]))
+        b1 = M2.act(np.array([0, 0, -l2 / 2]))
+        a2 = M1.act(np.array([0, 0, +l1 / 2]))
+        b2 = M2.act(np.array([0, 0, +l2 / 2]))
+        
+        ab = pinv((np.hstack((a1 - a2, b2 - b1))).reshape(3, 2)) * (b2 - a2)
```
Comment (jmirabel) [inline doc/d-practical-exercises/src/robot_hand.py:58]: Personally, I cannot read this line. Too many parenthesis.
Either
```
        ab = pinv( np.hstack( [a1 - a2, b2 - b1] ).reshape(3, 2) ) * (b2 - a2)
```
or
```
        ab = pinv( np.array( [ a1 - a2, b1 - b2 ]).T) ) * (b2 - a2)
```
or split it into multiple lines with intermediate variable(s).

**Review by jmirabel — COMMENTED** [review]
I think @jcarpent mentioned that you could use hpp-fcl to do this collision check rather than re-implementing it.

File: doc/d-practical-exercises/src/robot_hand.py (hunk 2 above)
Comment (kozakromch) [reply doc/d-practical-exercises/src/robot_hand.py]: I guess, it's done

**Comment by kozakromch** [comment]
@jmirabel Yes, jcarpent mentioned that I could use hpp-fcl . And I will use it in my project. Thank you for that.

This file is just a part of the exercise that I was happy to do because it allows me to understand how to use this library.
I think that the hpp-fcl should be mentioned in the exercise, but the exercise itself should be left as it is now.

File: doc/d-practical-exercises/src/robot_hand.py (hunk 2 above)
Comment (jmirabel) [reply doc/d-practical-exercises/src/robot_hand.py]: Thanks. Much clearer.

**Review by jmirabel — APPROVED** [review]

**Pull request merged**